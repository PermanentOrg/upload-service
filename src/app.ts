import express from "express";
import expressWinston from "express-winston";
import { apiRoutes } from "./routes";
import { logger } from "./log";

const app = express();
app.use(expressWinston.logger({ level: "http", winstonInstance: logger }));
app.use(express.json());
app.use("/api", apiRoutes);
export { app };
