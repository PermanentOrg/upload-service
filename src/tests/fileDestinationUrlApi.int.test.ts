import request from "supertest";
import { app } from "../app";

jest.mock("@aws-sdk/s3-presigned-post");
jest.mock("@aws-sdk/client-s3");

describe("fileDestinationUrl API #int", () => {
  const agent = request(app);
  describe("POST /api/fileDestinationUrl", () => {
    it("should specify all validation details when passed no parameters", async () => {
      const response = await agent.post("/api/fileDestinationUrl").expect(400);

      expect(response.body).toMatchSnapshot();
    });

    it("should specify all validation details when passed a negative maxSize", async () => {
      const response = await agent
        .post("/api/fileDestinationUrl")
        .send({
          maxSize: -1,
        })
        .expect(400);

      expect(response.body).toMatchSnapshot();
    });

    it("should return an S3 URL when passed the proper parameters", async () => {
      const response = await agent
        .post("/api/fileDestinationUrl")
        .send({
          bucket: "permanent-tests",
          path: "my_example_path",
          fileType: "image/png",
          fileName: "example.png",
          maxSize: 5000000,
        })
        .expect(200);

      expect(response.body).toMatchSnapshot();
    });
  });
});
