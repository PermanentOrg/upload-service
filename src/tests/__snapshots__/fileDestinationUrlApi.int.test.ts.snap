// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileDestinationUrl API #int POST /api/fileDestinationUrl should return an S3 URL when passed the proper parameters 1`] = `
{
  "destinationUrl": "https://permanent-tests.s3.amazonaws.com/my_example_path/example.png",
}
`;

exports[`fileDestinationUrl API #int POST /api/fileDestinationUrl should specify all validation details when passed a negative maxSize 1`] = `
{
  "error": {
    "_original": {
      "maxSize": -1,
    },
    "details": [
      {
        "context": {
          "key": "bucket",
          "label": "bucket",
        },
        "message": ""bucket" is required",
        "path": [
          "bucket",
        ],
        "type": "any.required",
      },
      {
        "context": {
          "key": "fileType",
          "label": "fileType",
        },
        "message": ""fileType" is required",
        "path": [
          "fileType",
        ],
        "type": "any.required",
      },
      {
        "context": {
          "key": "maxSize",
          "label": "maxSize",
          "limit": 1,
          "value": -1,
        },
        "message": ""maxSize" must be greater than or equal to 1",
        "path": [
          "maxSize",
        ],
        "type": "number.min",
      },
    ],
  },
}
`;

exports[`fileDestinationUrl API #int POST /api/fileDestinationUrl should specify all validation details when passed no parameters 1`] = `
{
  "error": {
    "_original": {},
    "details": [
      {
        "context": {
          "key": "bucket",
          "label": "bucket",
        },
        "message": ""bucket" is required",
        "path": [
          "bucket",
        ],
        "type": "any.required",
      },
      {
        "context": {
          "key": "fileType",
          "label": "fileType",
        },
        "message": ""fileType" is required",
        "path": [
          "fileType",
        ],
        "type": "any.required",
      },
      {
        "context": {
          "key": "maxSize",
          "label": "maxSize",
        },
        "message": ""maxSize" is required",
        "path": [
          "maxSize",
        ],
        "type": "any.required",
      },
    ],
  },
}
`;
